homeassistant/components/august/__init__.py

```diff
@@ -352,6 +359,12 @@ def update_lock_status(self, lock_id, lock_status, update_start_time_utc):
         self._lock_last_status_update_time_utc_by_id[lock_id] = update_start_time_utc
         return True
 
+    def lock_has_doorsense(self, lock_id):
+        """Determine if a lock has doorsense installed and can tell when the door is open or closed."""
+        # We do not update here since this is not expected
+        # to change until restart
+        return self._lock_detail_by_id[lock_id].doorsense
```

Comment:
`self._lock_detail_by_id[lock_id]` can still be set to `None` inside `_update_locks_detail`

Reply:
Added a check to see if its None